File: 2275/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(largestCombination([]int{8,8}))
}

func largestCombination(candidates []int) int {
	m := func(k int) int {
		res := 0
		for _, num := range candidates {
			if num & (1 << k) != 0 {
				res++
			}
		}
		return res
	}
	res := 0 
	for i := 0; i < 24; i++ {
		res = max(res, m(i))
	}
	return res
}

File: 2275/main.rs
struct Solution{}

impl Solution {
    pub fn largest_combination(candidates: Vec<i32>) -> i32 {
        fn m(candidates: &Vec<i32>, k: i32) -> i32 {
            let mut res = 0;
            for &num in candidates.iter() {
                if num & (1 << k) != 0 {
                    res += 1;
                }
            }
            res
        }
        let mut res = 0;
        for i in 0..24 {
            res = res.max(m(&candidates, i));
        }
        res
    }
}

fn main() {
    println!("{:#?}", Solution::largest_combination(vec![8,8]));
}